enterprise/backend/src/metabase_enterprise/advanced_permissions/models/permissions/general_permissions.clj

```diff
@@ -0,0 +1,126 @@
+(ns metabase-enterprise.advanced-permissions.models.permissions.general-permissions
+  "Code for generating and updating the General Permission graph. See [[metabase.models.permissions]] for more
+  details and for the code for generating and updating the *data* permissions graph."
+  (:require [clojure.data :as data]
+            [metabase.api.common :as api :refer [*current-user-id*]]
+            [metabase.models :refer [GeneralPermissionsRevision Permissions PermissionsGroup]]
+            [metabase.models.general-permissions-revision :as g-perm-revision]
+            [metabase.models.permissions :as perms]
+            [metabase.util.honeysql-extensions :as hx]
+            [metabase.util.schema :as su]
+            [schema.core :as s]
+            [toucan.db :as db]))
+
+;;; ---------------------------------------------------- Schemas -----------------------------------------------------
+
+(def ^:private GeneralPermissions (s/enum :yes :no))
+
+(def ^:private GroupPermissionsGraph
+  {:setting      GeneralPermissions
+   :monitoring   GeneralPermissions
+   :subscription GeneralPermissions
+   s/Keyword     GeneralPermissions})
+
+(def ^:private GeneralPermissionsGraph
+  {:revision s/Int
+   :groups   {su/IntGreaterThanZero GroupPermissionsGraph}})
+
+;; -------------------------------------------------- Fetch Graph ---------------------------------------------------
+
+(defn- group-id->permissions-set
+  "Get a map of general permissions for all existing group-ids"
+  []
+  (let [group-ids           (db/select-field :id PermissionsGroup)
+        general-permissions (db/select Permissions
+                                       {:where [:or
+                                                [:= :object "/"]
+                                                [:like :object (hx/literal "/general/%")]]})
+        permissions-set-from-group-id (fn [group-id]
+                                        (->> general-permissions
+                                             (filter #(= (:group_id %) group-id))
+                                             (map :object)
+                                             set))]
+    (into {} (for [group-id group-ids]
+               {group-id (permissions-set-from-group-id group-id)}))))
```

Comment:
Can't you do this with `group-by`, like the implementation of the function of the same name in `metabase.models.collection.graph`?

Reply:
No, the reason `metabase.models.collection.graph/group-id->permissions-set` will return a map of all group id that has `permissions`, That map might not contain a `group` if that group has no permissions. 

With my implentation, it's guaranteed that the function will return a map for all Permission groups, whether a Permission Group has permission or not

Reply:
that's fine, though, right? if the group doesn't show up in the graph, that just means it has no permissions. 

we don't enforce that the data perms graph has every group in it, for example

Reply:
then I think it's the question where do get the source of all `groups`, is it the graph itself, or from somewhere else?

maybe @alxnddr could help me answer this?
The question is: should a group with no general permissions show up in the graph?

My answer is it should.

Reply:
@qnkhuat both options work but the approach Noah described is already implemented for data and collection permission graphs. It also helps to transfer fewer data between BE and FE. What do you think about sticking to this approach?

Reply:
if you're okay with it then I'll go with what Noah described. 
I was worried that if the graph doesn't show a group, then FE will not aware of it, but seems like it's not the case